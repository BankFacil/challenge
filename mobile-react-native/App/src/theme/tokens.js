export const colors = {
	primary: '#515151',
	white: '#F7F7F7',
	border: '#BEBEBE'
};

export const space = [0, 4, 8, 16, 32, 64, 128, 256, 512];

export const radii = [0, 4, 8, 16, 32, 64, 128, 256, 512];

export const fontSizes = {
	sm: 12,
	md: 16,
	lg: 18,
};

export const fontWeights = {
	hairline: '100',
	thin: '200',
	light: '300',
	normal: '400',
	medium: '500',
	semibold: '600',
	bold: '700',
	extrabold: '800',
	black: '900',
  };
  