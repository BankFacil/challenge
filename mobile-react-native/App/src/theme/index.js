import * as tokens from './tokens';

export default {
  name: 'Github Theme',
  ...tokens,
};