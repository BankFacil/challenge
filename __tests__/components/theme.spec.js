import theme from '../../src/components/theme';

describe('theme', () => {
  describe('constants', () => {
    it('should return colors ', () => {
      const colors = theme.colors;
      expect(colors).toMatchSnapshot();
    });
    it('should return spacing ', () => {
      const spacing = theme.spacing;
      expect(spacing).toMatchSnapshot();
    });
  });

  describe('getColor', () => {
    it('should return color ', () => {
      const color = theme.getColorFromProps({secondary: true});
      expect(color).toMatch(theme.colors.secondary);
    });
    it('should return default color ', () => {
      const color = theme.getColorFromProps({});
      expect(color).toMatch(theme.colors.primary);
    });
  });
});
