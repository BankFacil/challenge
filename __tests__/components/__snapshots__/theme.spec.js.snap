// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`theme constants should return colors  1`] = `
Object {
  "dark": "#333",
  "error": "#FF612B",
  "grey": "#e2e2e2",
  "light": "#fff",
  "primary": "#3099DD",
  "secondary": "#FFC82B",
  "success": "#26E288",
}
`;

exports[`theme constants should return spacing  1`] = `
Object {
  "margin": 12,
  "padding": 12,
}
`;
